internal/host/statemachine.go

```diff
@@ -30,13 +30,32 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 			HostStatusKnown,
 			HostStatusDisconnected,
 			HostStatusInsufficient,
-			HostStatusResetting,
 			stateswitch.State(models.HostStatusResettingPendingUserAction),
 		},
 		DestinationState: HostStatusDiscovering,
 		PostTransition:   th.PostRegisterHost,
 	})
 
+	// Register host
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRegisterHost,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusResetting),
+		},
+		Condition:        th.IsHostInReboot,
+		DestinationState: stateswitch.State(models.HostStatusResetting),
```

Comment:
Why not move it to " resetting-pending-user-action"

Reply:
because then the cluster monitor won't set all the hosts to this status. 
cluster monitors checks if there's a host in status resetting and stage >= rebooting
if so it sets all the hosts to resetting-pending-user-action
if we change that here the other hosts won't change to resetting-pending-user-action.

Reply:
Consider adding a comment about the expected outcome of this transition (not straight forward) and why  it's required

Reply:
done